diff --git a/Cargo.toml b/Cargo.toml
index d331d58..9336742 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,7 +47,7 @@ raw-cpuid = "11.0"
 [target.'cfg(target_arch = "x86_64")'.dependencies]
 raw-cpuid = "11.0"
 
-[target.'cfg(not(any(target_os = "windows", target_arch = "wasm32")))'.dependencies]
+[target.'cfg(any(target_os = "emscripten", not(any(target_os = "windows", target_arch = "wasm32"))))'.dependencies]
 libc = "0.2"
 
 [target.'cfg(target_os = "windows")'.dependencies]
diff --git a/src/clocks/monotonic/mod.rs b/src/clocks/monotonic/mod.rs
index 80cbf89..7702014 100644
--- a/src/clocks/monotonic/mod.rs
+++ b/src/clocks/monotonic/mod.rs
@@ -13,7 +13,7 @@ mod wasm_wasi;
 #[cfg(all(target_arch = "wasm32", target_os = "wasi"))]
 pub use self::wasm_wasi::Monotonic;
 
-#[cfg(not(any(target_os = "windows", target_arch = "wasm32")))]
+#[cfg(any(target_os = "emscripten", not(any(target_os = "windows", target_arch = "wasm32"))))]
 mod unix;
-#[cfg(not(any(target_os = "windows", target_arch = "wasm32")))]
+#[cfg(any(target_os = "emscripten", not(any(target_os = "windows", target_arch = "wasm32"))))]
 pub use self::unix::Monotonic;
